contracts/solidity/NFTXLPStaking.sol

```diff
@@ -137,9 +140,11 @@ contract NFTXLPStaking is PausableUpgradeable {
             // We do this instead of patching in the token because
             // the xSLP contracts as currently deployed are not upgradeable.
             xSLPToken.timelockMint(msg.sender, amount, currentTimelock-block.timestamp);
+            emit Deposit(vaultId, pool.stakingToken, amount, msg.sender, currentTimelock-block.timestamp);
```

Comment:
`currentTimelock-block.timestamp` should avoid again calculating it by storing this value in cache as an uint